File: docs/guides/php/src/setSettingsThenSaveObjects.php
<?php

require __DIR__.'/../vendor/autoload.php';
use Algolia\AlgoliaSearch\Api\SearchClient;
use Algolia\AlgoliaSearch\Model\Search\IndexSettings;

$playlists = [/* Your records */];

$getAppIDFor = function ($user) {
    return ''; // Implement your own logic here
};

$getIndexingApiKeyFor = function ($user) {
    return ''; // Implement your own logic here
};

foreach ($playlists as $playlist) {
    try {
        // Fetch from your own data storage and with your own code
        // the associated application ID and API key for this user
        $appID = $getAppIDFor($playlist['user']);
        $apiKey = $getIndexingApiKeyFor($playlist['user']);

        $client = SearchClient::create($appID, $apiKey);

        $settings = [
            (new IndexSettings())
                ->setAttributesForFaceting(['filterOnly(user)']),
        ];
        $client->setSettings(
            '<YOUR_INDEX_NAME>',
            $settings,
        );

        $client->saveObjects(
            '<YOUR_INDEX_NAME>',
            $playlists,
        );
    } catch (Exception $e) {
        echo $e->getMessage().PHP_EOL;
    }
}

File: docs/guides/ruby/setSettingsThenSaveObjects.rb
require "algolia"

def get_app_id_for(_user)
  # Implement your own logic here
  ""
end

def get_indexing_api_key_for(_user)
  # Implement your own logic here
  ""
end

# Your records
playlists = []

playlists.each do |playlist|
  begin
    app_id = get_app_id_for(playlist["user"])
    api_key = get_indexing_api_key_for(playlist["user"])

    client = Algolia::SearchClient.create(app_id, api_key)

    settings = Algolia::Search::IndexSettings.new(
      attributes_for_faceting: ["filterOnly(user)"]
    )
    client.set_settings("<YOUR_INDEX_NAME>", settings)

    client.save_objects("<YOUR_INDEX_NAME>", playlists)
  rescue Exception => e
    puts("An error occurred: #{e}")
  end
end
